### DISCUSSION TITLE
How to Obtain Protein-Level Quantification from .parquet Files in DIA-NN 1.9.1 for Differential Analysis?

### DISCUSSION THREAD

**Discussion by YingtingPu** [vdemichev/DiaNN#1172] [Q&A] [answered]
Hi Vadim,

  I am new to proteomics data analysis. Recently, I used DIA-NN 1.9.1 to process my data and obtained several result tables. According to the software's README file, the .parquet file is the primary file for further analysis, but in the table, each protein may correspond to multiple precursor entries. My question is, how can I obtain the relative quantification results for proteins from the .parquet file, such as generating a results table similar to pg_matrix.tsv? I understand that it's not as simple as just summing the relative quantification values of the precursors, but I'm unclear about the exact criteria for extracting protein quantification information. Also, which software can be used for this, and what are the specific steps? Alternatively, can the pg_matrix.tsv table be directly used for downstream differential analysis?

**Comment by vdemichev** [comment, marked as answer]
Hi,

The protein quantities are already calculated in PG.MaxLFQ column, i.e. can just use those.

We suggest to use R (or Python) to work with the main report in .parquet format. This is eventually the best, also using R saves you a lot of time with making plots for papers, etc. Takes 1 day to 1 week to master the basics of R (which is more than enough) without any prior programming experience.

With R, you can get a matrix like pg_matrix with code like this:
```
library(arrow)
library(reshape2)

# pivot data frame to obtain features x runs table
pivot <- function(df, file.header, id.header, quantity.header) {
  df <- unique(df[,c(file.header, id.header, quantity.header)])
  melted <- as.data.frame(melt(df, id.vars = c(file.header, id.header), measure.vars = c(quantity.header)))
  melted$value[which(melted$value == 0)] <- NA
  piv <- dcast(melted, as.formula(paste0(id.header,'~',file.header)), value.var = "value") 
  rownames(piv) <- piv[[1]]
  piv[,1] <- NULL
  as.matrix(piv)
}

data <- read_parquet('report.parquet') # put the report file name & location here
df <- data[data$Q.Value <= 0.01 & data$PG.Q.Value <= 0.05 & data$Lib.Q.Value <= 0.01 & data$Lib.PG.Q.Value <= 0.01,] # with MBR
# df <- data[data$Q.Value <= 0.01 & data$PG.Q.Value <= 0.05 & data$Global.Q.Value <= 0.01 & data$Global.PG.Q.Value <= 0.01,] # without MBR

pg <- pivot(df, 'Run', 'Protein.Group', 'PG.MaxLFQ')
```

If you don't want to use R or Python but just Excel, then please just use pg_matrix directly, it's designed to be optimal or near-optimal for a wide range of experiments.

Best,
Vadim

**Reply by sooheon** [reply]
For mbr, why do you filter for both lib and original q values? Doesn't mbr mean you should use the mbr'd library (lib) preferentially, and ignore the original set of q values?

**Reply by vdemichev** [reply]
MBR search is the 'same as non-MBR', just with a DIA-based library which is automatically generated. So all basic q-value filters should still stay.

**Reply by YingtingPu** [reply]
Hi Vadim,

Thanks for getting back to me.
After seeing your response, I followed your instructions and processed the data in R, successfully obtaining the quantification results. This is really exciting, and I truly appreciate your help. I also noticed that the processed results are almost identical to the pg_matrix output from the software itself. Next, I plan to proceed with further analysis based on the results obtained from R. However, before moving forward, I'd like to confirm with you whether my main steps for obtaining results using DIA-NN are correct:

First, I downloaded the protein database in fasta format from the UniProt database and imported it into DIA-NN to generate the predicted spectral library.
Next, I imported my raw data, along with the predicted spectral library and the previously mentioned UniProt protein database, into DIA-NN to obtain recommended values for MS1, MS2, and scan window parameters.
Finally, I adjusted the parameters according to the recommended values and ran it again to get the final results.

Best regards,
YingtingPu

**Reply by vdemichev** [reply]
Hi YingtingPu,

Sounds all good!

Best,
Vadim
